File: Ahorn/triggers/ResetFlagsTrigger.jl
module XaphanHelperResetFlagsTrigger

using ..Ahorn, Maple

@mapdef Trigger "XaphanHelper/ResetFlagsTrigger" ResetFlagsTrigger(x::Integer, y::Integer, setTrueFlags::String="", setFalseFlags::String="", transitionUpdate::Bool=false, removeWhenOutside::Bool=false, registerInSaveData::Bool=false)

const placements = Ahorn.PlacementDict(
    "Reset Flags Trigger (Xaphan Helper)" => Ahorn.EntityPlacement(
        ResetFlagsTrigger,
        "rectangle"
    )
)

end

File: Loenn/triggers/ResetFlagsTrigger.lua
local ResetFlagsTrigger = {}

ResetFlagsTrigger.name = "XaphanHelper/ResetFlagsTrigger"
ResetFlagsTrigger.fieldOrder = {
    "x", "y", "width", "height", "setTrueFlags", "setFalseFlags", "conditionFlags", "transitionUpdate", "removeWhenOutside", "registerInSaveData", "inverted"
}
ResetFlagsTrigger.placements = {
    name = "ResetFlagsTrigger",
    data = {
        setTrueFlags = "",
        setFalseFlags = "",
        transitionUpdate = false,
        removeWhenOutside = false,
        registerInSaveData = false,
        conditionFlags = "",
        inverted = false
    }
}

return ResetFlagsTrigger